### ISSUE TITLE
文件比较多的时候无法打包，是不是我设置不对

### ISSUE THREAD

**Issue by somnusnox** [ccloli/E-Hentai-Downloader#292]
这是出现错误的日志部分，打包试了几次总是最后卡住然后出现错误日志
[EHD] Index > 1350  | RealIndex > 1703  | Name > 205.jpg  | RetryCount > 0  | DownloadedCount > 1348  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1349  | RealIndex > 1702  | Name > 204.jpg  | RetryCount > 0  | DownloadedCount > 1349  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1351  | RealIndex > 1704  | Name > 206.jpg  | RetryCount > 0  | DownloadedCount > 1350  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1348  | RealIndex > 1701  | Name > 203.jpg  | RetryCount > 0  | DownloadedCount > 1351  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1352  | RealIndex > 1705  | Name > 207.jpg  | RetryCount > 0  | DownloadedCount > 1352  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1353  | RealIndex > 1706  | Name > 208.jpg  | RetryCount > 0  | DownloadedCount > 1353  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1355  | RealIndex > 1708  | Name > 210.jpg  | RetryCount > 0  | DownloadedCount > 1354  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1356  | RealIndex > 1709  | Name > 211.jpg  | RetryCount > 0  | DownloadedCount > 1355  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1354  | RealIndex > 1707  | Name > 209.jpg  | RetryCount > 0  | DownloadedCount > 1356  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1357  | RealIndex > 1710  | Name > 212.jpg  | RetryCount > 0  | DownloadedCount > 1357  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1358  | RealIndex > 1711  | Name > 213.jpg  | RetryCount > 0  | DownloadedCount > 1358  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1359  | RealIndex > 1712  | Name > 214.jpg  | RetryCount > 0  | DownloadedCount > 1359  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1362  | RealIndex > 1715  | Name > 217.jpg  | RetryCount > 0  | DownloadedCount > 1360  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1364  | RealIndex > 1717  | Name > 219.jpg  | RetryCount > 0  | DownloadedCount > 1361  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1360  | RealIndex > 1713  | Name > 215.jpg  | RetryCount > 0  | DownloadedCount > 1362  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1361  | RealIndex > 1714  | Name > 216.jpg  | RetryCount > 0  | DownloadedCount > 1363  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1363  | RealIndex > 1716  | Name > 218.jpg  | RetryCount > 0  | DownloadedCount > 1364  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1366  | RealIndex > 1719  | Name > 221.jpg  | RetryCount > 0  | DownloadedCount > 1365  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1368  | RealIndex > 1721  | Name > 223.jpg  | RetryCount > 0  | DownloadedCount > 1366  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1367  | RealIndex > 1720  | Name > 222.jpg  | RetryCount > 0  | DownloadedCount > 1367  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1365  | RealIndex > 1718  | Name > 220.jpg  | RetryCount > 0  | DownloadedCount > 1368  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1369  | RealIndex > 1722  | Name > 224.jpg  | RetryCount > 0  | DownloadedCount > 1369  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1373  | RealIndex > 1726  | Name > 229.jpg  | RetryCount > 0  | DownloadedCount > 1370  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1371  | RealIndex > 1724  | Name > 227.jpg  | RetryCount > 0  | DownloadedCount > 1371  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1370  | RealIndex > 1723  | Name > 225.jpg  | RetryCount > 0  | DownloadedCount > 1372  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1372  | RealIndex > 1725  | Name > 228.jpg  | RetryCount > 0  | DownloadedCount > 1373  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1377  | RealIndex > 1730  | Name > 233.jpg  | RetryCount > 0  | DownloadedCount > 1374  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1374  | RealIndex > 1727  | Name > 230.jpg  | RetryCount > 0  | DownloadedCount > 1375  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1375  | RealIndex > 1728  | Name > 231.jpg  | RetryCount > 0  | DownloadedCount > 1376  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1376  | RealIndex > 1729  | Name > 232.jpg  | RetryCount > 0  | DownloadedCount > 1377  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1378  | RealIndex > 1731  | Name > 234.jpg  | RetryCount > 0  | DownloadedCount > 1378  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1379  | RealIndex > 1732  | Name > 235.jpg  | RetryCount > 0  | DownloadedCount > 1379  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1383  | RealIndex > 1736  | Name > 239.jpg  | RetryCount > 0  | DownloadedCount > 1380  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1380  | RealIndex > 1733  | Name > 236.jpg  | RetryCount > 0  | DownloadedCount > 1381  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1384  | RealIndex > 1737  | Name > 240.jpg  | RetryCount > 0  | DownloadedCount > 1382  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1382  | RealIndex > 1735  | Name > 238.jpg  | RetryCount > 0  | DownloadedCount > 1383  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1381  | RealIndex > 1734  | Name > 237.jpg  | RetryCount > 0  | DownloadedCount > 1384  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:14632 [EHD] Request Image Limits From e-hentai.org
E-Hentai Downloader.user.js:14660 [EHD] Image Limits > {"timestamp":1719330383007,"cur":"26542","total":"50000","donatorPower":7}
E-Hentai Downloader.user.js:12532 [EHD] Index > 1386  | RealIndex > 1739  | Name > 242.jpg  | RetryCount > 0  | DownloadedCount > 1385  | FetchCount > 5  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1385  | RealIndex > 1738  | Name > 241.jpg  | RetryCount > 0  | DownloadedCount > 1386  | FetchCount > 4  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1388  | RealIndex > 1741  | Name > 244.jpg  | RetryCount > 0  | DownloadedCount > 1387  | FetchCount > 3  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1389  | RealIndex > 1742  | Name > 245.jpg  | RetryCount > 0  | DownloadedCount > 1388  | FetchCount > 2  | FailedCount > 0
E-Hentai Downloader.user.js:12532 [EHD] Index > 1387  | RealIndex > 1740  | Name > 243.jpg  | RetryCount > 0  | DownloadedCount > 1389  | FetchCount > 1  | FailedCount > 0
E-Hentai Downloader.user.js:14632 [EHD] Request Image Limits From e-hentai.org
E-Hentai Downloader.user.js:14660 [EHD] Image Limits > {"timestamp":1719330389823,"cur":"26566","total":"50000","donatorPower":7}
E-Hentai Downloader.user.js:12674 Uncaught (in promise) ReferenceError: abData is not defined
    at errorHandler (E-Hentai Downloader.user.js:12674:10)
E-Hentai Downloader.user.js:14632 [EHD] Request Image Limits From e-hentai.org
E-Hentai Downloader.user.js:14660 [EHD] Image Limits > {"timestamp":1719330443007,"cur":"26558","total":"50000","donatorPower":7}
E-Hentai Downloader.user.js:14852 [EHD] Pre-calculate estimated cost > {"cost":23886,"normalCost":23886,"leastCost":1742,"perCost":1,"gp":42546,"isUsingOriginal":true,"isUsingGP":false}
E-Hentai Downloader.user.js:12674 Uncaught (in promise) ReferenceError: abData is not defined
    at errorHandler (E-Hentai Downloader.user.js:12674:10)
E-Hentai Downloader.user.js:14632 [EHD] Request Image Limits From e-hentai.org
E-Hentai Downloader.user.js:14660 [EHD] Image Limits > {"timestamp":1719330863011,"cur":"26489","total":"50000","donatorPower":7}
E-Hentai Downloader.user.js:12674 Uncaught (in promise) ReferenceError: abData is not defined
    at errorHandler (E-Hentai Downloader.user.js:12674:10)
errorHandler @ E-Hentai Downloader.user.js:12674
E-Hentai Downloader.user.js:14632 [EHD] Request Image Limits From e-hentai.org
E-Hentai Downloader.user.js:14660 [EHD] Image Limits > {"timestamp":1719331163013,"cur":"26439","total":"50000","donatorPower":7}
E-Hentai Downloader.user.js:12674 Uncaught (in promise) ReferenceError: abData is not defined
    at errorHandler (E-Hentai Downloader.user.js:12674:10)
errorHandler @ E-Hentai Downloader.user.js:12674

**Comment by ccloli** [comment]
See https://github.com/ccloli/E-Hentai-Downloader?tab=readme-ov-file#memory-usage

tl;dr, 请使用 Pages Range 分多次下载，或使用 gallery-dl 等非在浏览器内执行的工具

**Comment by somnusnox** [comment]
感谢你的回应，这几天忘记回复了，抱歉~不过我想知道卡住会有明确的图片数量和大小上限么，还是说只要多了会随缘卡住，不清楚限制范围？




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ccloli/E-Hentai-Downloader"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年6月27日(星期四) 凌晨0:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ccloli/E-Hentai-Downloader] 文件比较多的时候无法打包，是不是我设置不对 (Issue #292)





 
See https://github.com/ccloli/E-Hentai-Downloader?tab=readme-ov-file#memory-usage
 
tl;dr, 请使用 Pages Range 分多次下载，或使用 gallery-dl 等非在浏览器内执行的工具

**Comment by ccloli** [comment]
建议按照整个 gallery 的大小与页面数计算下文件的平均大小，一般来说不超过 500MB 为佳

如果 gallery 过大，建议优先使用 torrents 或官方自己的打包下载服务。看你是有捐赠的，理论上会有免费的下载额度

**Comment by somnusnox** [comment]
感谢你回我~torrents这是那个种子链接我知道，官方自己的打包下载服务是什么呢，我捐倒是给他捐满了，但是页面好像也没看到有其他可点击的选项，可以请你给我讲讲么？




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ccloli/E-Hentai-Downloader"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月1日(星期一) 晚上11:11
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ccloli/E-Hentai-Downloader] 文件比较多的时候无法打包，是不是我设置不对 (Issue #292)





 
建议按照整个 gallery 的大小与页面数计算下文件的平均大小，一般来说不超过 500MB 为佳
 
如果 gallery 过大，建议优先使用 torrents 或官方自己的打包下载服务。看你是有捐赠的，理论上会有免费的下载额度

**Comment by ccloli** [comment]
在 Gallery 的右上方有一个 Archive Download 的链接，点击后会有个弹窗，当确认下载后，服务器会自动将所有文件打包为一个 zip 文件。一般来说下载会需要花费 GP 或 credit，不过对于捐赠用户，你应该是有免费额度的。关于官方功能的具体介绍，可参考 https://ehwiki.org/wiki/Archives/Chinese

看你的 Donator Power 为 7，应该对应 Catgirl 这个档，那么理论上你每天都会有 100GB 的从官方服务器下载的额度。关于捐赠档位的具体功能，可参考 https://ehwiki.org/wiki/Donation/Chinese

**Comment by somnusnox** [comment]
你对规则好熟悉啊，那我明天看看，谢谢你为我解惑，晚安~




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ccloli/E-Hentai-Downloader"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月1日(星期一) 晚上11:29
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ccloli/E-Hentai-Downloader] 文件比较多的时候无法打包，是不是我设置不对 (Issue #292)





 
在 Gallery 的右上方有一个 Archive Download 的链接，点击后会有个弹窗，当确认下载后，服务器会自动将所有文件打包为一个 zip 文件。一般来说下载会需要花费 GP 或 credit，不过对于捐赠用户，你应该是有免费额度的。关于官方功能的具体介绍，可参考 https://ehwiki.org/wiki/Archives/Chinese
 
看你的 Donator Power 为 7，应该对应 Catgirl 这个档，那么理论上你每天都会有 100GB 的从官方服务器下载的额度。关于捐赠档位的具体功能，可参考 https://ehwiki.org/wiki/Donation/Chinese
